tensorflow_addons/layers/wrappers.py

```diff
@@ -182,13 +180,56 @@ def _init_norm(self):
             g_tensor = self.g.assign(tf.reshape(v_norm, (self.layer_depth,)))
             return [g_tensor]
 
+    def _calculate_moments(self, x, axes, keep_dims=True):
```

Comment:
Can we ask to expose this in Tensorflow directly?

Reply:
I think Tensorflow has a too complex test system. (In my other pull request, I had a very hard time.)
And also, its framework always doesn't have backward compatibility in particular with Multi-GPU/TPU.

So I have no enough power for sending the pull request for TensorFlow...

Reply:
I don't know if you could push more effort here cause https://github.com/tensorflow/addons/pull/1789#issuecomment-625876871. /cc @gabrieldemarmiesse

Reply:
Yeah, we need to rethink the implementation of `WeightNormalization`. To the best of my knowledge it's impossible to hook ourselves because the use of a weight in a layer without using hacky stuff that will come back to bite us later.

We'll likely have to deprecate the wrapper layer and provide utility functions instead, with a good tutorial to explain how the use need to do the subclassing for it to work. We could maybe provide a new Dense layer with the weight normalization implemented inside. That would cover many use cases and provide an example of how to do it for other layers.

It would be nice if somebody could work on this and provide a proposal/possibly a new API to fix all those issues.